### ISSUE TITLE
Spec has incorrect signature for requestReferenceSpace

### ISSUE THREAD

**Issue by NellWaliczek** [immersive-web/webxr#543]
The explainer document describes the `requestReferenceSpace()` API as accepting an array of reference space descriptions, but the spec is written to only accept one.
Instead of:
```webidl
Promise<XRReferenceSpace> requestReferenceSpace(XRReferenceSpaceOptions options
);
```

It should be:
```webidl
  Promise<XRReferenceSpace> requestReferenceSpace(Array<XRReferenceSpaceOptions> preferredOptions);
```

**Comment by ddorwin** [comment]
Please do make this soon as it may change how we look at the API (i.e., in #514).

The parameter name should probably indicate that this is a list of supported option bundles. Something like supportedConfigs. ("supportedOptions" is a bit ambiguous since it is a list of option definitions rather than just a list of options.)

**Comment by ddorwin** [comment]
On a related note, is "Options" the right term for `XRReferenceSpaceOptions` and similar types? It seems to be used to define a specific configuration. For example, `requestReferenceSpace()`, accepts a list of configurations that are acceptable.

I'm not sure which term is most common in web APIs. EME uses "Configuration." getUserMedia() uses "Constraints," but that is specific to that API design.

**Comment by ddorwin** [comment]
FYI, `Array` is not a Web IDL type. I think `sequence` is probably correct here.

**Comment by toji** [comment]
Just added #549 to correct this, but in the process of doing so I started to question whether or not we should.

For one, I can't imagine ever having more than two elements in this array, and only in a couple of very specific scenarios: A `bounded` reference space falling back to a `stationary/floor-level` one, and MAYBE an `unbounded` space falling back to a `stationary/eye-level`. Everything else is too semantically different to treat as a blind fallback. Keeping the interface this way also necessitates keeping the object types separate so that you can tell which of the options array was accepted attribute, as opposed to combining them as suggested in #513.

All in all, I'm not convinced that benefit outweighs the ability to avoid structuring your code like so (which I don't find all that offensive to begin with):

```js
session.requestReferenceSpace({ type: 'bounded' }).then(onRefSpace)
    .catch(() => { session.requestReferenceSpace({ type: 'stationary', subtype: 'floor-level' }).then(onRefSpace);
```

**Comment by NellWaliczek** [comment]
Hey @TrevorFSmith!  This convenience was added at your suggestion :) How strongly to do you feel about keeping it?

**Comment by toji** [comment]
After feedback from the most recent IWWG call, merged #554 which closes this issue by bringing the explainer signature inline with the spec. (Removing the ability to pass in an array of options)
